src/KineticConvolution/Point.cs

```diff
@@ -1,32 +1,80 @@
 using System;
 
-namespace KineticConvolution
+namespace Hilke.KineticConvolution
 {
     public class Point : IEquatable<Point>
     {
+        public Point(IAlgebraicNumber x, IAlgebraicNumber y)
+        {
+            X = x ?? throw new ArgumentNullException(nameof(x));
+            Y = y ?? throw new ArgumentNullException(nameof(y));
+        }
+
         public IAlgebraicNumber X { get; }
 
         public IAlgebraicNumber Y { get; }
 
-        public Point(IAlgebraicNumber x, IAlgebraicNumber y)
+        /// <inheritdoc />
+        public bool Equals(Point? other)
         {
-            X = x ?? throw new ArgumentNullException(nameof(x));
-            Y = y ?? throw new ArgumentNullException(nameof(y));
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return X.Equals(other.X)
```

Comment:
Just before the `return`, we should also test the types.

Reply:
why? Class is sealed.